lib/ansible/plugins/action/template.py

```diff
@@ -62,6 +62,8 @@ def run(self, tmp=None, task_vars=None):
         dest   = self._task.args.get('dest', None)
         force  = boolean(self._task.args.get('force', True))
         state  = self._task.args.get('state', None)
+        decode = self._task.args.pop('decode', None)
+        encode = self._task.args.pop('encode', None)
```

Comment:
One of my previous implementation was using utf-8 but then, I get crash in unit test. So I decided to  use a None value and to use old code path in order to respect the old behaviour.
We can change this to utf-8 but we need to change python3 tests.